Firestore/core/test/firebase/firestore/local/leveldb_local_store_test.cc

```diff
@@ -25,25 +28,48 @@ namespace {
 
 class TestHelper : public LocalStoreTestHelper {
  public:
+  TestHelper(std::unique_ptr<QueryEngine> query_engine, bool index_free)
+      : query_engine_(std::move(query_engine)), index_free_(index_free) {
+  }
+
   std::unique_ptr<Persistence> MakePersistence() override {
     return LevelDbPersistenceForTesting();
   }
 
+  QueryEngine* query_engine() override {
+    return query_engine_.get();
+  }
+
   /** Returns true if the garbage collector is eager, false if LRU. */
   bool IsGcEager() const override {
     return false;
   }
+
+  bool IsIndexFree() const override {
```

Comment:
`query_engine()->type() == QueryEngine::Type::IndexFree` is how callers should be able to test this. You should be able to remove this.

Reply:
Yes, removed.